ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -193,6 +193,11 @@ data instance BlockQuery (ShelleyBlock proto era) :: Type -> Type where
     -> BlockQuery (ShelleyBlock proto era)
                   (SL.PState (EraCrypto era))
 
+  GetPoolDistr
+    :: Maybe (Set (SL.KeyHash 'SL.StakePool (EraCrypto era)))
+    -> BlockQuery (ShelleyBlock proto era)
+                  (SL.PoolDistr (EraCrypto era))
```

Comment:
@JaredCorduan @lehins I think I asked a similar question about `PState` on a recent&similar PR, but this time it's about `PoolDistr`:

Do we want to expose `PoolDistr` in the query API? That's promising some stability in it's interface.

Reply:
I think it is okay, but I do not know for sure.

Our definition of "big" (ie the data structures that we know will be finding a home one day on disk) is "at least the size of the number of registered stake credentials". The `PoolDistr` is a consolidation of the per-stake credential stake distribution, which is itself not "big", but is derived from a "big" structure. I am currently expecting that we will keep the pool distribution in memory, and that this query will not be disrupted. But it is intimately related to a "big" data structure that we do know will be disrupted.